### ISSUE TITLE
Catalog Search module search with parenthesis breaks the site.

### ISSUE THREAD

**Issue by ron-jomashop** [magento/magento2#23159]
### Preconditions (*)

1. Magento 2.3.1

### Steps to reproduce (*)

1. Enable Elastic Search
2. append '/catalogsearch/result/?q=%28%28%28%28' to url. Ex: site.com/catalogsearch/result/?q=%28%28%28%28
3. The site will return a error.

### Expected result (*)

1. Should not have any errors.

### Actual result (*)

1. ```
2 exception(s):
Exception #0 (Zend_Db_Statement_Exception): SQLSTATE[42000]: Syntax error or access violation: 1064 syntax error, unexpected $end, query was: SELECT `search_synonyms`.* FROM `search_synonyms` WHERE (MATCH (synonyms) AGAINST ('\\(' IN BOOLEAN MODE))
Exception #1 (PDOException): SQLSTATE[42000]: Syntax error or access violation: 1064 syntax error, unexpected $end
```

**Comment by ron-jomashop** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @ron-jomashop. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @ron-jomashop, here is your [Magento instance](https://i-23159-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23159-2-3-develop.instances.magento-community.engineering/admin](https://i-23159-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by ron-jomashop** [comment]
It is reproducible on the Magento instance.

**Comment by ron-jomashop** [comment]
@magento give me 2.3.1 instance

**Comment by magento-engcom-team** [comment]
Hi @ron-jomashop. Thank you for your request. I'm working on Magento 2.3.1 instance for you

**Comment by magento-engcom-team** [comment]
Hi @ron-jomashop, here is your [Magento instance](https://i-23159-2-3-1.instances.magento-community.engineering).
Admin access: [https://i-23159-2-3-1.instances.magento-community.engineering/admin](https://i-23159-2-3-1.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by ron-jomashop** [comment]
Cannot produce the error. Close the ticket.

**Comment by ron-jomashop** [comment]
It is replicable with ElasticSearch enabled. I am using elasticsearch2 right now. Cannot reproduce on the instance because it doesn't have elasticsearch running in the background.

**Comment by davidverholen** [comment]
@ron-jomashop thanks for your report.

While I was able to reproduce the issue on a fresh 2.3.1 instance it seems to be fixed in the 2.3-develop branch. So it should already be fixed for 2.3.2

Since 2.2-develop does not support elasticsearch yet, this issue can be closed.

**Comment by ron-jomashop** [comment]
@magento-engcom-team give me 2.3.1 instance with elasticsearch

**Comment by magento-engcom-team** [comment]
Hi @ron-jomashop. Thank you for your request. I'm working on Magento 2.3.1 instance for you

**Comment by ron-jomashop** [comment]
@magento give me 2.3-develop instance with elasticsearch

**Comment by magento-engcom-team** [comment]
Hi @ron-jomashop. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @ron-jomashop, here is your [Magento instance](https://i-23159-2-3-1.instances.magento-community.engineering).
Admin access: [https://i-23159-2-3-1.instances.magento-community.engineering/admin](https://i-23159-2-3-1.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by ron-jomashop** [comment]
I tested it with 2.3.x develop version and it also has this bug.

**Comment by ron-jomashop** [comment]
> @ron-jomashop thanks for your report.
> 
> While I was able to reproduce the issue on a fresh 2.3.1 instance it seems to be fixed in the 2.3-develop branch. So it should already be fixed for 2.3.2
> 
> Since 2.2-develop does not support elasticsearch yet, this issue can be closed.

@davidverholen I have tested it with 2.3-develop and it is not working. Can you test it again? Thanks!

**Comment by davidverholen** [comment]
@ron-jomashop I tried again and could not reproduce it (while I was able to reproduce it directly on 2.3.1)

**Steps I tried to reproduce**:
1. Install vanilla 2.3-develop instance with sample data
2. start elastic search server via docker-compose (see config below)
3. log into admin, change search engine to elasticsearch 6.0+
4. clear cache
5. rebuild index
6. open `/catalogsearch/result/?q=%28`

**Environment**:
php: 7.2.13
magento: 2.3-develop
es server docker config:
```yaml
version: '2.2'
services:
  es01:
    image: elasticsearch:6.7.2
    container_name: magento2.es
    environment:
      - node.name=magento2.es
      - cluster.name=magento2.es
      - bootstrap.memory_lock=true
      - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
    ulimits:
      memlock:
        soft: -1
        hard: -1
    volumes:
      - esdata01:/usr/share/elasticsearch/data
    ports:
      - 9200:9200
    networks:
      - esnet


volumes:
  esdata01:
    driver: local

networks:
  esnet:
```

**Result**:
<img width="1264" alt="Bildschirmfoto 2019-06-07 um 01 04 23" src="https://user-images.githubusercontent.com/2813693/59071754-4d179400-88c0-11e9-8024-302e10284823.png">

**Comment by ron-jomashop** [comment]
> @ron-jomashop I tried again and could not reproduce it (while I was able to reproduce it directly on 2.3.1)
> 
> **Steps I tried to reproduce**:
> 
> 1. Install vanilla 2.3-develop instance with sample data
> 2. start elastic search server via docker-compose (see config below)
> 3. log into admin, change search engine to elasticsearch 6.0+
> 4. clear cache
> 5. rebuild index
> 6. open `/catalogsearch/result/?q=%28`
> 
> **Environment**:
> php: 7.2.13
> magento: 2.3-develop
> es server docker config:
> 
> ```yaml
> version: '2.2'
> services:
>   es01:
>     image: elasticsearch:6.7.2
>     container_name: magento2.es
>     environment:
>       - node.name=magento2.es
>       - cluster.name=magento2.es
>       - bootstrap.memory_lock=true
>       - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
>     ulimits:
>       memlock:
>         soft: -1
>         hard: -1
>     volumes:
>       - esdata01:/usr/share/elasticsearch/data
>     ports:
>       - 9200:9200
>     networks:
>       - esnet
> 
> 
> volumes:
>   esdata01:
>     driver: local
> 
> networks:
>   esnet:
> ```
> 
> **Result**:
> <img alt="Bildschirmfoto 2019-06-07 um 01 04 23" width="1264" src="https://user-images.githubusercontent.com/2813693/59071754-4d179400-88c0-11e9-8024-302e10284823.png">

I am sorry but can you test it with ElasticSearch 2 again? I think it only occurs on older version of elasticsearch.

**Comment by davidverholen** [comment]
@ron-jomashop now also tested with elasticsearch2 and it also worked. Please make sure you are on a vanilla magento and on the latest 2.3-develop when trying to reproduce it

**Comment by ron-jomashop** [comment]
Alright. I will look into it. Thanks!

**Comment by ron-jomashop** [comment]
It seems like this commit fixed the issue

https://github.com/magento/magento2/commit/ec7015017b1e0c27c161fa62b1ead3a6367271e8

**Comment by ron-jomashop** [comment]
@davidverholen The Magento fix will only works on one parentheses. If you search "((((", it will break even on latest version of the Magento 2.

**Comment by davidverholen** [comment]
@ron-jomashop thx for investigating. Could you maybe update the preconditions then we can reopen it for confirmation (I'm currently not on my work machine and cannot reproduce it here)

**Comment by ron-jomashop** [comment]
Steps to reproduce (*)
Enable Elastic Search
append '/catalogsearch/result/?q=%28%28%28%28' to url. Ex: site.com/catalogsearch/result/?q=%28%28%28%28
The site will return a error.

The reason why Magento2.3.2 works is because it skips the search when it is less than 3 characters.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @davidverholen 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-17777` were created

**Issue Available**: @davidverholen, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by ryanjavens** [comment]
@ron-jomashop I'm noticing a similar bug on one of my sites as well. For search queries that are still working on your site, did you notice any inconsistent search results before upgrading to 2.3.1? For example, searching for a product name is returning results in 2.2.x, but not in 2.3.x. Trying to determine if my two issues are related or not. Thanks!

**Comment by magento-engcom-team** [comment]
Hi @ron-jomashop. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-23783` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.
